src/components/reusable/daterangepicker/daterangepicker.stories.js

```diff
@@ -1,122 +1,420 @@
+              aria-label=${args._textStrings?.requiredText || 'Required'}
+              >*</abbr
+            >`
+          : null}
+        ${args.startDateLabel}
+      </label>
+      <kd-button
+        slot="start-anchor"
+        size="medium"
+        value=""
+        iconPosition="center"
+        kind="primary-app"
+        type="button"
+        size="medium"
+        ?disabled=${args.dateRangePickerDisabled}
+        description="Date range picker button containing icon"
+      >
+        <span slot="icon" style="line-height: 1;"
+          >${unsafeSVG(overflowIcon)}</span
+        >
+      </kd-button>
+    </kyn-date-range-picker>
+  `;
 };
 
-export const DateTimeRangePicker = {
-  args: { ...args, datePickerType: 'date-time' },
-  render: (args) => {
-    return html`
-      <kyn-date-range-picker
-        size=${args.size}
-        name=${args.name}
-        datePickerType=${args.datePickerType}
-        caption=${args.caption}
+const ButtonIconTextAnchorTemplate = (args) => {
+  useEffect(() => {
+    return () => {
+      disconnectFlatpickr();
+    };
+  }, []);
+
+  const startAnchorId = 'date-range-picker-start';
+
+  return html`
+    <kyn-date-range-picker
+      .nameAttr="${args.nameAttr}"
+      .locale="${args.locale}"
+      .dateFormat="${args.dateFormat}"
+      ?required="${args.required}"
+      .value="${args.value}"
+      .warnText="${args.warnText}"
+      .invalidText="${args.invalidText}"
+      .altFormat=${args.altFormat}
+      .disable="${args.disable}"
+      .enable="${args.enable}"
+      .multipleInputs="${args.multipleInputs}"
+      .caption="${args.caption}"
+      ?dateRangePickerDisabled="${args.dateRangePickerDisabled}"
+      ?twentyFourHourFormat="${args.twentyFourHourFormat}"
+      .minDate="${args.minDate}"
+      .maxDate="${args.maxDate}"
+      .startDateLabel="${args.startDateLabel}"
+      .endDateLabel="${args.endDateLabel}"
+      @on-change=${(e) => action(e.type)(e)}
+    >
+      <label
+        slot="start-label"
+        class="label-text"
+        for=${startAnchorId}
+        ?disabled=${args.dateRangePickerDisabled}
+      >
+        ${args.required
+          ? html`<abbr
+              class="required"
+              title=${args._textStrings?.requiredText || 'Required'}
+              aria-label=${args._textStrings?.requiredText || 'Required'}
+              >*</abbr
+            >`
+          : null}
+        ${args.startDateLabel}
+      </label>
+      <kd-button
+        slot="start-anchor"
+        value="Primary"
+        kind=${'primary-app'}
+        id=${startAnchorId}
+        class="btn interactive"
+        ?disabled=${args.dateRangePickerDisabled}
+      >
+        <span part="icon" style="margin: 3px 12px 0 0;"
+          >${unsafeSVG(calendarIcon)}</span
+        >
+        Choose Date Range
+      </kd-button>
+    </kyn-date-range-picker>
+  `;
+};
+
+const MultiInputTemplate = (args) => {
```

Comment:
I think we just need to go with one variant, probably single input.

Reply:
✅